### PR TITLE
feat: add eip-7685 requests root to header

### PR DESCRIPTION
**Pull request by onbjerg**
Adds EIP-7685 header fields to both the consensus and RPC `Header` types.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
all of this looks correct, this basically shifts replaces the parent beacon root as the last entry

pending @Rjected 

I assume we also need additional checks in the consensus impl?

**Comment by onbjerg** [comment]
@mattsse yeah i'm opening PRs as I progress on https://github.com/paradigmxyz/reth/pull/8053, we need to add some fields to body + calc the root itself, will keep the alloy impl on par

**Comment by prestwich** [comment]
should we be managing changes in e.g. an enum over versions?

**Comment by mattsse** [comment]
> should we be managing changes in e.g. an enum over versions?

while this would have advantages, we'd still need this type for decoding any rlp blob or we need additional context for decoding, like the hardfork so we know how to decode

**Comment by prestwich** [comment]
great i love it when field semantics change without any structural indication 🎉

File: crates/consensus/src/header.rs [hunk 1]
```diff
@@ -285,15 +291,22 @@ impl Header {
...
         // need to be repeated and placeholder length added. Otherwise, it's impossible to
         // tell _which_ fields are missing. This is mainly relevant for contrived cases
         // where a header is created at random, for example:
         //  * A header is created with a withdrawals root, but no base fee. Shanghai blocks are
         //    post-London, so this is technically not valid. However, a tool like proptest would
         //    generate a block like this.
-        if let Some(parent_beacon_block_root) = self.parent_beacon_block_root {
-            length += parent_beacon_block_root.length();
+        if let Some(requests_root) = self.requests_root {
+            length += requests_root.length();
```
Comment (Rjected) [inline crates/consensus/src/header.rs:309]: note: the hack in this comment is not needed in reth any more, the arbitrary impl was changed instead, we should copy it over so the encoding / decoding impl is more sensible

**Review by Rjected — APPROVED** [review]
lgtm

File: crates/consensus/src/header.rs (hunk 1 above)
Comment (onbjerg) [reply crates/consensus/src/header.rs:309]: will do

**Pull request merged**